AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine+Resolver.swift

```diff
@@ -24,6 +24,8 @@ extension RemoteSyncEngine {
 
             case (.initializeSubscriptions, .initializedSubscriptions):
                 return .performInitialSync
+            case (.initializeSubscriptions, .errored(let error)):
+                return .cleanup(error)
```

Comment:
I needed to find a way to pass the error from the completion to the retry block (since it goes from completion -> .errored/finished -> scheduleRestart) so passing error through these enum values was one way to do this.  If there's a better way to do this, i'm all ears.